pkg/agent/util/net.go

```diff
@@ -106,47 +108,63 @@ func dialUnix(address string) (net.Conn, error) {
 	return net.Dial("unix", address)
 }
 
-// GetIPNetDeviceFromIP returns a local IP/mask and associated device from IP.
-func GetIPNetDeviceFromIP(localIP net.IP) (*net.IPNet, *net.Interface, error) {
+// GetIPNetDeviceFromIP returns local IPs/masks and associated device from IP.
+func GetIPNetDeviceFromIP(localIPs *ip.IPs) (v4IPNet *net.IPNet, v6IPNet *net.IPNet, iface *net.Interface, err error) {
 	linkList, err := net.Interfaces()
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	for _, link := range linkList {
-		addrList, err := link.Addrs()
+	for i := range linkList {
+		addrList, err := linkList[i].Addrs()
 		if err != nil {
 			continue
 		}
 		for _, addr := range addrList {
```

Comment:
Probably add some comments to say the list should include at most one IPv4 and one IPv6, so the double loops should be work.

Reply:
Added comments.